backend/app/infrastructure/__init__.py

```diff
@@ -30,8 +32,12 @@ def _add_address_canonization(handler):
             def _decorated(*args, **kwargs):
                 field_value = kwargs.get(field_name)
                 if force or field_value is not None:
-                    updated_field = to_checksum_address(field_value)
-                    kwargs.update([(field_name, updated_field)])
+                    try:
+                        updated_field = to_checksum_address(field_value)
+                    except ValueError:
+                        raise InvalidAddressFormat()
+                    else:
+                        kwargs.update([(field_name, updated_field)])
```

Comment:
IIUC currently this canonization is being applied if field_name is in ['user_address', 'project_address']. Greping through code, I see other field names that can get this treatment: 'address', 'userAddress', 'proposalAddress', 'projectAddress'.

Reply:
It's used for all address-related field names that comes as an input to our API so it should be covered too